File: modified_scripts/r/binary_search/qwen2.5-coder_3b.r
```r
binary_search <- function(arr, x) {
  low <- 1
  high <- length(arr)
  
  while (low <= high) {
    mid <- low + (high - low) %/% 2
    
    if (arr[mid] == x) {
      return(mid)
    }
    
    if (arr[mid] < x) {
      low <- mid + 1
    } else {
      high <- mid - 1
    }
  }
  
  return(-1)
}
```

File: modified_scripts/julia/binary_search/qwen2.5-coder_3b.jl
```julia
function binary_search(arr, target)
    low = 1
    high = length(arr)

    while low <= high
        mid = div(low + high, 2)

        if arr[mid] == target
            return mid
        elseif arr[mid] < target
            low = mid + 1
        else
            high = mid - 1
        end
    end

    return -1
end
```